### PR TITLE
Throw an exception if we generate a partition with zero cells

### PR DESCRIPTION
**Pull request by wdhawkins**
Throw an error instead of crashing if we generate a partition with zero cells. This can easily happen if we have more ranks than cells. For us to exit gracefully in this case, it's necessary to have every location calculate min/max/avg cells per partition and throw if min is zero.

### CODE REVIEW

File: framework/mesh/mesh_generator/mesh_generator.cc [hunk 1]
```diff
@@ -177,31 +198,13 @@ MeshGenerator::PartitionMesh(const UnpartitionedMesh& input_umesh, int num_parti
     }
   }
 
-  // Note A: We do not add the diagonal here. If we do it, ParMETIS seems
+  // Note A: We do not add the diagonal here. If we do, ParMETIS seems
```
Comment (ragusa) [inline framework/mesh/mesh_generator/mesh_generator.cc:201]: I know it is not your comment, but what does this mean? And what about PTSCOTCH then? Should we create an issue to remember to investigate the meaning of that comment? It seems important for parallel efficiency.

**Comment by andrsd** [comment]
Should partition with zero cells really be a stopping condition?

File: framework/mesh/mesh_generator/mesh_generator.cc (hunk 1 above)
Comment (wdhawkins) [reply framework/mesh/mesh_generator/mesh_generator.cc:201]: I'm not sure what this comment means. I'm not even sure it's still relevant to the code that it references. It doesn't look to me like we are excluding anything. I'll open an issue to investigate what's going on here.

**Comment by wdhawkins** [comment]
> Should partition with zero cells really be a stopping condition?

I believe it should. I think from a "know what you're doing and use the code correctly" perspective, it always should. But, I can see the argument for robustness. I think there would be a number of fixes that would be necessary to support this. With a few minor fixes, I can get get the code into the topological sorting for the sweep, but then things go sideways again. At some point, I'll probably rewrite that chunk of code, and I'll try to make it work for cases where we have partitions with no cells.

**Review by zhardy-lanl — APPROVED** [review]

**Review by andrsd — APPROVED** [review]

File: framework/mesh/mesh_generator/mesh_generator.cc (hunk 1 above)
Comment (ragusa) [reply framework/mesh/mesh_generator/mesh_generator.cc:201]: see email

**Review by ragusa — APPROVED** [review]

**Pull request merged**